controllers/suite_util_test.go

```diff
@@ -311,6 +311,17 @@ func waitDoneMessageOnComponent(componentKey types.NamespacedName) {
 }
 
 func expectPacBuildStatus(componentKey types.NamespacedName, state string, errID int, errMessage string, mergeURL string) {
+	// should first wait for desired state, as there is small time window when
+	// message might be still done from previous run, but even though action already finished, it didn't update status yet
+	// so it will get status from previous run
```

Comment:
Regarding the "previous run", do you mean a previous test in same context might be still running while current test is running?

Reply:
no, but previous test ran and finished and set
message=done
and pac state=enabled

problem is that when testing failed build
we can wait for finilazer to be gone
message = done
but there is still time between that and actually updating component with new build status

so waiting at least on desired state will help with the race